src/app/components/pages/settings/pending-transactions/pending-transactions.component.ts

```diff
@@ -41,4 +59,28 @@ export class PendingTransactionsComponent implements OnInit {
       return transaction;
     });
   }
+
+  private getWalletsTransactions(transactions: any): Observable<any> {
+    const allTransactions = this.getUpdatedTransactions(transactions);
+
+    return Observable.zip(allTransactions, this.walletService.all, (trans: any, wallets: Wallet[]) => {
+      const walletAddresses = [];
```

Comment:
I think the procedure would be more efficient `walletAddresses` a Map instead of an array.